pkg/api/v1/atlasproject_types.go

```diff
@@ -295,6 +298,22 @@ func (p *AtlasProject) WithMaintenanceWindow(window project.MaintenanceWindow) *
 	return p
 }
 
+func (p *AtlasProject) LastSpecFrom(annotation string) (*AtlasProjectSpec, error) {
```

Comment:
I would really like us to avoid attaching methods in the public API surface. This nicely fits into the translation layer.

Reply:
It cannot be on translation layer because is to be used on the sub-resources, that are not using translation layer, at least PEs are not. Anyway, I will move it to the controller instead.

Reply:
In that case let's at least have this method **not** in the public API surface.